util/tracecpu/mock/mock.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mock
+
+import (
+	"sync"
+	"time"
+
+	"github.com/pingcap/parser"
+	"github.com/pingcap/tidb/util/tracecpu"
+	"github.com/uber-go/atomic"
+)
+
+// TopSQLCollector uses for testing.
+type TopSQLCollector struct {
+	sync.Mutex
+	// sql_digest -> normalized SQL
+	sqlMap map[string]string
+	// plan_digest -> normalized plan
+	planMap map[string]string
+	// (sql + plan_digest) -> sql stats
+	sqlStatsMap map[string]*tracecpu.SQLStats
+	collectCnt  atomic.Int64
+}
+
+// NewTopSQLCollector uses for testing.
+func NewTopSQLCollector() *TopSQLCollector {
+	return &TopSQLCollector{
+		sqlMap:      make(map[string]string),
+		planMap:     make(map[string]string),
+		sqlStatsMap: make(map[string]*tracecpu.SQLStats),
+	}
+}
+
+// Collect uses for testing.
+func (c *TopSQLCollector) Collect(ts int64, stats []tracecpu.SQLStats) {
+	defer c.collectCnt.Inc()
+	if len(stats) == 0 {
+		return
+	}
+	c.Lock()
+	defer c.Unlock()
+	for _, stmt := range stats {
+		hash := c.hash(stmt)
+		stats, ok := c.sqlStatsMap[hash]
+		if !ok {
+			tmp := stmt
+			stats = &tmp
```

Comment:
The first statement's `CPUTimeMs` is calculated twice? it's started from the first non-exist, rather than 0

Reply:
Great catch, already fix it.